exercises/zebra-puzzle/.meta/src/reference/java/ZebraPuzzle.java

```diff
@@ -0,0 +1,108 @@
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Stream;
+
+class ZebraPuzzle {
+    private String waterDrinker = "";
+    private String zebraOwner = "";
+
+    ZebraPuzzle() {
+        this.solve();
+    }
+
+    private static boolean isJustRightOf(int a, int b) {
+        return a - 1 == b;
+    }
+
+    private static boolean nextTo(int a, int b) {
+        return isJustRightOf(a, b) || isJustRightOf(b, a);
+    }
+
+    private void solve() {
+        int[] houses = new int[]{1,2,3,4,5};
+        int first = 1;
+        int middle = 3;
+
+        perms(houses).forEach(colors -> {
+            int red = colors[0], green = colors[1], ivory = colors[2], yellow = colors[3], blue = colors[4];
```

Comment:
I'd prefer each variable to be declared on a new line. That makes it more obvious how many variables there are, and some of the lines below that declare many variables like this are very long 🙂